backend/src/StamAcasa.Api/Controllers/ProfileController.cs

```diff
@@ -165,5 +166,21 @@ public async Task<IActionResult> UpdateFamilyProfile(int id, [FromBody] UserProf
 
             return new OkObjectResult(result != null);
         }
+
+        /// <summary>
+        /// Delete user profile and related data.
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete]
+        public async Task<IActionResult> DeleteProfile()
+        {
+            var sub = User.Claims.FirstOrDefault(c => c.Type == "sub")?.Value;
+            if (sub == null)
+                return new UnauthorizedResult();
+
+            await _userService.DeleteUserAndDependentData(sub);
```

Comment:
Why don't we get the user email from the current session instead of requiring the UI to provide the email ?

Reply:
Yup, I think this is a very fair point.